### PR TITLE
Revisit release config

### PR DESCRIPTION
**Pull request by pheenomenon**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] No Release

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,10 @@
         "mocha": "^7.1.2",
         "mock-require": "^3.0.3",
         "nock": "^12.0.3",
-        "nyc": "^15.0.1"
+        "nyc": "^15.0.1",
+        "semantic-release": "17.0.7",
+        "@semantic-release/git": "9.0.0",
+        "conventional-changelog-eslint": "3.0.8"
```
Comment (alexkli) [inline package.json:25]: What does `conventional-changelog-eslint` do?

Comment (tmathern) [reply package.json:25]: Additional commit message parsing?

Comment (jdelbick) [reply package.json:25]: https://www.npmjs.com/package/conventional-changelog-eslint

IIUC, this checks our commit messages to make sure they are understandable by semantic release...aka so we don't accidentally forget to release, etc.

Comment (pheenomenon) [reply package.json:25]: semantic-release uses the commit messages to determine the type of changes in the codebase. By default it uses Angular Commit Message Conventions.  - 
Reference: https://github.com/semantic-release/semantic-release#commit-message-format

It lets us configure the commit conventions as a preset, in which case it must be installed.
Reference: https://github.com/semantic-release/commit-analyzer#options

For now, I have kept the current convention very simple. 

`By default for any commit message there will be no release.
Only if someone wants a release, they have to say it in commit message by **prefixing** 
- if they want patch(BUGFIX-RELEASE:) , minor(FEATURE-RELEASE:) or major(BREAKING-RELEASE:)  in the commit message. `

**Pull request merged**